### PR TITLE
[ticket/15752] Add events to storage class

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-15752

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15752

### CODE REVIEW

File: phpBB/phpbb/storage/storage.php [hunk 1]
```diff
@@ -106,7 +115,28 @@ protected function get_adapter()
...
 	public function put_contents($path, $content)
 	{
-		$this->get_adapter()->put_contents($path, $content);
+		$adapter = $this->get_adapter();
+		$storage = $this->get_name();
+
+		/**
+		* Before writing in the file
+		*
+		* @event core.storage_put_contents_before
```
Comment (Nicofuma) [inline phpBB/phpbb/storage/storage.php]: you can use multiple dots if you want (`core.storage.put_contents_before` would make sense per example)

Comment (rubencm) [reply phpBB/phpbb/storage/storage.php]: None of the events use multiple dots

Comment (Nicofuma) [reply phpBB/phpbb/storage/storage.php]: well, both are okayish

File: phpBB/phpbb/storage/storage.php [hunk 2]
```diff
@@ -112,7 +121,28 @@ public function put_contents($path, $content)
...
+		* @var	\phpbb\storage\adapter\adapter_interface	adapter	Adapter
+		* @var	string	storage										Storage name
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: Why not call it storage_name then?

File: phpBB/phpbb/storage/storage.php [hunk 3]
```diff
@@ -112,7 +121,28 @@ public function put_contents($path, $content)
...
+		* @var	string	path										Path to file
+		* @var	array	content										Data
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: File data

File: phpBB/phpbb/storage/storage.php [hunk 4]
```diff
@@ -134,7 +164,26 @@ public function get_contents($path)
...
-		return $this->get_adapter()->get_contents($path);
+		$adapter = $this->get_adapter();
+		$storage = $this->get_name();
+
+		/**
+		* Before read the file
+		*
+		* @event core.storage_get_contents_before
+		* @var	\phpbb\storage\adapter\adapter_interface	adapter	Adapter
+		* @var	string	storage										Storage name
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: Same as above

File: phpBB/phpbb/storage/storage.php [hunk 5]
```diff
@@ -147,7 +196,26 @@ public function get_contents($path)
...
+		* @since 3.3.0-a1
+		*/
+		$vars = array(
+			'adapter',
+			'storage',
+			'path',
+		);
+		extract($this->dispatcher->trigger_event('core.storage_exists_before', compact($vars)));
+
+		return ($this->is_tracked($path) && (!$full_check || $adapter()->exists($path)));
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: Should be `$adapter->exists($path)`

File: phpBB/phpbb/storage/storage.php [hunk 6]
```diff
@@ -147,7 +196,26 @@ public function get_contents($path)
...
-		return ($this->is_tracked($path) && (!$full_check || $this->get_adapter()->exists($path)));
+		$adapter = $this->get_adapter();
+		$storage = $this->get_name();
+
+		/**
+		* Before check if file exist
+		*
+		* @event core.storage_exists_before
+		* @var	\phpbb\storage\adapter\adapter_interface	adapter	Adapter
+		* @var	string	storage										Storage name
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: See above

File: phpBB/phpbb/storage/storage.php [hunk 7]
```diff
@@ -165,7 +233,26 @@ public function delete($path)
...
-		$this->get_adapter()->delete($path);
+		$adapter = $this->get_adapter();
+		$storage = $this->get_name();
+
+		/**
+		* Before delete the file
+		*
+		* @event core.storage_delete_before
+		* @var	\phpbb\storage\adapter\adapter_interface	adapter	Adapter
+		* @var	string	storage										Storage name
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: See above

File: phpBB/phpbb/storage/storage.php (hunk 2 above)
Comment (rubencm) [reply phpBB/phpbb/storage/storage.php]: how?

File: phpBB/phpbb/storage/storage.php (hunk 5 above)
Comment (rubencm) [reply phpBB/phpbb/storage/storage.php]: Uhm, that error isn't from this pr

**Pull request closed without merging**